lib/req-res/responder.js

```diff
@@ -39,42 +40,48 @@ util.inherits( Responder, Rabbus.Responder );
 // Instance Methods
 // ----------------
 Responder.prototype.inbound = function ( message, respond, callback ) {
+	const timeout = config.timeout.set( {
+		respond, message,
+		'log'         : this.logger,
+		'emitter'     : this,
+		'messageType' : this.messageType
+	} );
+
 	const statusOptions = {
-		respond,
-		message,
+		respond, message, timeout,
 		'log'         : this.logger,
 		'messageType' : this.messageType,
 		'emitter'     : this
 	};
 
-	if ( !this.vSchema || this.vSchema === 'undefined' ) {
-		// if no validation, invoke callback immediately
-		callback( message, requestStatus( statusOptions ) );
-	} else {
-		// perform validation if there is a validation schema
-		validate( {
-			'value'   : message,
-			'schema'  : this.vSchema,
-			'options' : this.vOptions
-		} )
-		.then( function ( data ) {
-			callback( data, requestStatus( statusOptions ) );
-		} )
-		.catch( function ( error ) {
-			// send fail
-			respond( {
-				'status' : 'fail',
-				'data'   : error.message
-			} );
-		} );
-	}
+	const send = requestStatus( statusOptions );
+
+	// perform validation if there is a validation schema
+	validate( {
+		'value'   : message,
+		'schema'  : this.vSchema,
+		'options' : this.vOptions
+	} )
+	.then( function ( data ) {
+		callback( data, send );
+	} )
+	.catch( Err.UnknownSchemaError, () => {
+		// backward compatibility for no validation
+		callback( message, send );
+	} )
+	.catch( Err.ValidationError, ( validationErr ) => {
+		send.fail( validationErr.message );
+		return clearTimeout( statusOptions.timeout );
+	} )
+	.catch( function ( error ) {
+		send.error( error.message, error );
```

Comment:
It looks like we can just do `send.error( error.message, error.stack || error );` and it will show up fine in logstash.
![screenshot 2016-04-20 09 17 14](https://cloud.githubusercontent.com/assets/419983/14679828/b87f8eb8-06d8-11e6-87f3-a6d6ab429e64.png)